engine/function/src/templates/Numeric.ftl

```diff
@@ -1268,38 +1268,63 @@ public class Numeric {
 
         if (n == 0) {
             return NULL_DOUBLE;
-        } else {
-            // NULL values sorted to beginning of the array.
-            ${pt.primitive}[] copy = sort(values.toArray());
+        }
+
+        ${pt.primitive}[] sorted = values.copyToArray();
+        Arrays.sort(sorted);
 
-            // Determine if there are any NULL in the array.
-            int nullCount = 0;
-            for (int i = 0; i < n; i++) {
-                if (isNull(copy[i])) {
+        int nullStart = -1;
+        int nullCount = 0;
+
+        <#if pt.valueType.isFloat >
```

Comment:
There are slightly different impls for float vs. integral iteration because we need to check every float value for NaN but we can exit early for integrals. It might be unneeded optimization...